Task_2/src/components/Products/ProductPage.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import ProductModel from "../../Models/ProductModel";
+import { AiOutlinePercentage } from "react-icons/ai";
+import { BsCurrencyDollar } from "react-icons/bs";
+import { GoStarFill } from "react-icons/go";
+import { FaRegHeart } from "react-icons/fa";
+import { RiShoppingCartLine } from "react-icons/ri";
+import { IoMdShare } from "react-icons/io";
+import ProductGridImage from "./ProductImageGrig";
+import { useParams } from "react-router-dom";
+
+const ProductPage: React.FC = () => {
+  const Prodcut: ProductModel = {
+    id: 0,
+    title: "",
+    description: "",
+    price: 0,
+    discountPercentage: 0,
+    rating: 0,
+    stock: 0,
+    brand: "",
+    category: "",
+    thumbnail: "",
+    images: [],
+  };
+  const {productId} = useParams();
+  console.log(productId)
+
+  const [data, setData] = useState(Prodcut);
+
+  useEffect(() => {
+    const fetchData = async () => {
```

Comment:
Not a correct way to call API. useEffect hook runs after the rendering of the entire UI or component completes. So when we put an API call in it, the API call will start after the complete rendering of UI. Instead use **_useCallback_** 

calling the API inside **_useCallback_** ensures that the function remains stable across re-renders, preventing unnecessary API calls and re-renders unless the dependencies of the **_useCallback_** change

Reply:
i need more time for this solution, stuck with errors.